spec/unit/type/rpmkey_spec.rb

```diff
@@ -71,5 +71,16 @@
       end
     end
 
+    describe "autorequire" do
+      it "should autorequire a local file" do
+        catalog = Puppet::Resource::Catalog.new
+        file = Puppet::Type.type(:file).new(:name => '/tmp/foo', :content => 'bar' )
+        catalog.add_resource file
+        key = described_class.new(:name => 'DB42A60E', :source => '/tmp/foo', :ensure => :present)
+        catalog.add_resource key
+        key.autorequire.size.should == 1
+      end
+    end
```

Comment:
No, we do not have to test the autorequire functionality itself, I was more thinking about the regex working properly. But you are right in a way that even if the regex was faulty and matched just anything, the autorequire array would still be empty because we'd have no matching resource in the catalog.